fedimintd/src/fedimintd.rs

```diff
@@ -470,8 +470,9 @@ fn check_release_notes_ack() -> anyhow::Result<()> {
     const FM_SKIP_REL_NOTES_ACK_ENV: &str = "FM_SKIP_REL_NOTES_ACK";
     const FM_REL_NOTES_ACK_ENV: &str = "FM_REL_NOTES_ACK";
     // the suffix here is to prevent people getting to smart and trying to
-    // automate-it-out change it for every release
-    const FM_REL_NOTES_CUR_VAL: &str = "0_4_xyz";
+    // automate-it-out
+    // update to match the current release
+    const FM_REL_NOTES_CUR_VAL: &str = "0_4_0";
```

Comment:
![image](https://github.com/user-attachments/assets/7e5310dc-7d32-486a-982d-d6f4c9d49042)

Reply:
Oh interesting, I read that as "change it for every release"

I'm not sure I fully understand the solution here since you can still set `FM_SKIP_REL_NOTES_ACK_ENV` to automate the process of acking for each release.

Reverting

Reply:
> you can still set FM_SKIP_REL_NOTES_ACK_ENV to automate the process of acking for each release.

:shushing_face: That's not documented anywhere. Or at least in official docs. :D

Reply:
Since we'd only expect `FM_SKIP_REL_NOTES_ACK_ENV` to be enabled in test envs, could we instead use `is_running_in_test_env` to skip the ack step, that way users can't automate out the ack?

Also, you've likely already thought of this, but new federations starting from 0.4 will still need to set the upgrade ack even though it isn't necessary to have a coordinated shutdown.

Reply:
>  that way users can't automate out the ack

If someone really, really, really need it, then I'd like an escape hatch. I just want to avoid telling people about it.

> will still need to set the upgrade ack even though it isn't necessary to have a coordinated shutdown.

Yes. I think it's OK. Maybe we should add a line about it in the docs. "If you are setting a new Federation with fedimintd 0.4, you don't need to do anything unusual, but you still need to acknowledge reading these release notes."

@bradleystachurski

Reply:
If I'm understanding correctly, the logic of `check_release_notes_ack` means that for all versions going forward they'll need to set an ack var until we remove or update `check_release_notes_ack`.

Perhaps we can check the data dir from the server opts to see if this is a new fedimintd? I'm a bit concerned about the UX for first time users from all versions going forward.

Out-of-scope for this PR so I'll create a new issue to track.

Reply:
https://github.com/fedimint/fedimint/issues/5674

Reply:
I am hoping that by 0.5 we could get consensus version voting, and that should eliminate the need for the lock step upgrades for most changes going forward.